### ISSUE TITLE
Allow specifying labels for the `helm.k0sproject.io/v1beta1` objects

### ISSUE THREAD

**Issue by laverya** [k0sproject/k0s#4364]
### Is your feature request related to a problem? Please describe.

Currently, the helmchart objects have one label - `k0s.k0sproject.io/stack: helm`. It would make it easier for us to backup and restore these objects if we were able to add additional labels to the object.

### Describe the solution you would like

There was a top-level field within 'helmextensions' that allowed specifying a set of labels to apply to every chart.

### Describe alternatives you've considered

Backing up with the `k0s.k0sproject.io/stack: helm` label should be possible, but isn't as easy to configure.

**Comment by twz123** [comment]
You can already do arbitrary things with Chart resources if you manage them as plain manifests by placing yaml files into `<data-dir>/manifests` instead of putting them into the k0s config directly.

**Comment by laverya** [comment]
That would work great if we weren't using [Dynamic Configuration](https://docs.k0sproject.io/stable/dynamic-configuration/)! Though I imagine you'd recommend just creating chart objects in the cluster ourselves in that scenario? It would simplify some other logic of ours I suppose

**Comment by twz123** [comment]
> Though I imagine you'd recommend just creating chart objects in the cluster ourselves in that scenario? It would simplify some other logic of ours I suppose

Yes, this is more versatile. The charts specified in the k0s config aren't any special compared to user-managed chart resources. And if you're using dynamic config already, then it's potentially not too hard to reconcile additional chart resources in addition to the k0s cluster config resource, I suppose.
